fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/IndexOptions.java

```diff
@@ -223,6 +224,143 @@ public class IndexOptions {
      */
     public static final String RTREE_USE_NODE_SLOT_INDEX = "rtreeUseNodeSlotIndex";
 
+    /**
+     * HNSW-only: The random seed that is used to probabilistically determine the highest layer of an insert into an
+     * HNSW structure. See {@link Config#getRandomSeed()}. The default random seed is
+     * {@link Config#DEFAULT_RANDOM_SEED}.
+     */
+    public static final String HNSW_RANDOM_SEED = "hnswRandomSeed";
+
+    /**
+     * HNSW-only: The metric that is used to determine distances between vectors. The default metric is
+     * {@link Config#DEFAULT_METRIC}. See {@link Config#getMetric()}.
+     */
+    public static final String HNSW_METRIC = "hnswMetric";
+
+    /**
+     * HNSW-only: The number of dimensions used. All vectors must have exactly this number of dimensions. This option
+     * must be set when interacting with a vector index as it there is no default.
+     * See {@link Config#getNumDimensions()}.
+     */
+    public static final String HNSW_NUM_DIMENSIONS = "hnswNumDimensions";
+
+    /**
+     * HNSW-only: Indicator if all layers except layer {@code 0} use inlining. If inlining is used, each node is
+     * persisted as a key/value pair per neighbor which includes the vectors of the neighbors but not for itself. If
+     * inlining is not used, each node is persisted as exactly one key/value pair per node which stores its own vector
+     * but specifically excludes the vectors of the neighbors. The default value is set to
+     * {@link Config#DEFAULT_USE_INLINING}. See {@link Config#isUseInlining()}.
```

Comment:
I wonder if this should have a bit more "why". (Or alternatively, if `Config#isUseInlining` should explain it.) Because this tells me what the option controls, but not the tradeoff. I believe the tradeoff is that the inlining node uses more space, but it allows the relevant information for searching a layer to be read in fewer round trips, decreasing query latency.

I don't know how much to workshop this, at this moment. I suspect what we actually want here is to do a bit more benchmarking and then come up with a recommendation, and that spending too much time on the docs at this moment may be counter productive.

Reply:
I added explanation to `Config.isUseInlining()`.